mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -193,7 +193,7 @@ func resourceSqlDatabaseInstance() *schema.Resource {
 								Schema: map[string]*schema.Schema{
 									"bucket": {
 										Type:         schema.TypeString,
-										Required:     true,
+										Optional:     true,
```

Comment:
Would you mind adding / modifying an existing tests to test the `sql_server_audit_config` with `bucket` specified. If that works, we may want to add at `AtLeastOneOf` to all these three subfields to prevent empty block config. Example of how to use `AtLeastOneOf`: https://googlecloudplatform.github.io/magic-modules/docs/how-to/update-handwritten-resource/#implement-the-respective-flattener-andor-expander-for-the-new-field

Reply:
There is already a test which sets the bucket field. Did you mean only the bucket field?
Also, I think we should allow empty block config here since the API is falsely returning this block for non-SQL Server instances. If we use `AtLeastOneOf`, that would mean the user is required to provide one of these fields.

Reply:
What I mean about test is to test `sql_server_audit_config` being configured without `bucket` being specified as to test if the `bucket` field is optional.
> Also, I think we should allow empty block config here since the API is falsely returning this block for non-SQL Server instances. If we use `AtLeastOneOf`, that would mean the user is required to provide one of these fields.
Would you mind explaining a bit how `allowing empty block config` will solve the issue `the API is falsely returning this block for non-SQL Server instances`? It sounds like an API error but not Terraform one? Sorry if I got confused.
`AtLeastOneOf` just means one of the subfields should be provided only if users have `sql_server_audit_config` set up. If they don't need this, they can omit this block from their configs.

Reply:
Sorry for the confusion as I was under the wrong impression. We can include `AtLeastOneOf` here.